src/package-hoister.js

```diff
@@ -121,7 +121,42 @@ export default class PackageHoister {
         return sortAlpha(aPattern, bPattern);
       });
 
-      for (const [pattern, parent] of queue) {
+      // sort the queue again to hoist packages without peer dependencies first
+      let sortedQueue = [];
+      const availableSet = new Set();
+
+      for (let hasChanged = true; queue.length > 0 && hasChanged;) {
+        hasChanged = false;
+
+        for (let t = 0; t < queue.length; ++t) {
+          const pattern = queue[t][0];
+          const pkg = this.resolver.getStrictResolvedPattern(pattern);
+
+          const peerDependencies = Object.keys(pkg.peerDependencies || {});
+          const areDependenciesFulfilled = peerDependencies.every(peerDependency => availableSet.has(peerDependency));
```

Comment:
>Additionally, peer dependencies are rare enough for this code not to have a major (or even minor) impact on the perf. It's unlikely more than a single pass will be ever required.

I think this is a dangerous assumption in the long run since people are using `peerDependencies` more and more with many libraries as they are the "right way" for things like plugins etc.